### PR TITLE
Added resources requirements for the Connect S2I build

### PR DESCRIPTION
**Pull request by stanlyDoge**
Signed-off-by: Stanislav Knot <sknot@redhat.com>

### Type of change
- Enhancement / new feature

### Description
ssia

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Update/write design documentation in `./design`
- [x] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: documentation/book/appendix_crds.adoc [hunk 1]
```diff
@@ -1613,6 +1613,8 @@ Used in: xref:type-KafkaConnectS2I-{context}[`KafkaConnectS2I`]
 |integer
 |image                     1.2+<.<|The docker image for the pods.
 |string
+|buildResources            1.2+<.<|CPU and memory resources to reserve (limits and requests) for the build.
```
Comment (laidan6000) [inline documentation/book/appendix_crds.adoc]: I don't quite follow. Why is "limits and requests" in parentheses?

Comment (stanlyDoge) [reply documentation/book/appendix_crds.adoc]: It is copypaste from the description of other resources.

Comment (laidan6000) [reply documentation/book/appendix_crds.adoc]: Well, it doesn't make sense there either! Does "using limits and requests" make sense?

Comment (stanlyDoge) [reply documentation/book/appendix_crds.adoc]: What about "CPU and memory resources limits and requests for the ..."?

**Comment by scholzj** [comment]
@stanlyDoge Could you please rebase this so that it can be merged?

**Pull request merged**